src/app/pages/pages.module.ts

```diff
@@ -8,14 +8,26 @@ import { ShopListComponent } from './shop-list/shop-list.component';
 import { DeliveryComponent } from './delivery/delivery.component';
 import { SharedModule } from '../shared/shared.module';
 import { CommonModule } from '@angular/common';
+import { ProductComponent } from './product/product.component';
+import { FormsModule } from '@angular/forms';
 
 const routes: Routes = [
   { path: '', component: MainPageComponent },
   { path: 'about', component: AboutComponent },
-  { path: 'catalog', component: CatalogComponent },
+  {
+    path: 'catalog',
+    children: [
+      { path: '', component: CatalogComponent },
+      { path: ':id', component: ProductComponent },
+    ],
+  },
   { path: 'contacts', component: ContactsComponent },
   { path: 'shops', component: ShopListComponent },
   { path: 'delivery', component: DeliveryComponent },
+  {
+    path: '**',
+    redirectTo: 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa',
```

Comment:
Strange redirect

You probably should have error page or redirect to default page here instead